src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -785,14 +785,15 @@ BoundNode bindSpreadElement(BoundCollectionExpressionSpreadElement element, Type
             return element;
         }
 
-        internal bool TryGetCollectionIterationType(ExpressionSyntax syntax, TypeSymbol collectionType, out TypeWithAnnotations iterationType)
+        internal bool TryGetCollectionExpressionIterationType(ExpressionSyntax syntax, TypeSymbol collectionType, out TypeWithAnnotations iterationType)
         {
             BoundExpression collectionExpr = new BoundValuePlaceholder(syntax, collectionType);
             return GetEnumeratorInfoAndInferCollectionElementType(
                 syntax,
                 syntax,
                 ref collectionExpr,
                 isAsync: false,
+                allowExtensionMethod: false, // collection expression target types do not require extension method GetEnumerator
```

Comment:
>allowExtensionMethod: false, // collection expression target types do not require extension method GetEnumerator [](http://example.com/codeflow?start=16&length=112)

I suggest to not add this parameter. It looks like we can simply check ```builder.ViaExtensionMethod``` when `GetEnumeratorInfoAndInferCollectionElementType` succeeds.